File: API/API/repositorys/IProjectRepository.vb
Imports System.Collections.Generic

Public Interface IProjectRepository
    Function GetById(id As Guid) As Project
    Function Add(project As Project) As Project
    Sub Delete(project As Project)
    Function GetProjectsByUserId(userId As Guid) As List(Of Project)
    Function GetProjectMembers(projectId As Guid) As List(Of ProjectMember)
    Function GetMember(projectId As Guid, userId As Guid) As ProjectMember
    Sub AddMember(member As ProjectMember)
    Sub RemoveMember(member As ProjectMember)
    Sub Save()
End Interface

File: API_v2/Repositorys/IRepositorys/IProjectRepository.cs
using API_v2.Models;

namespace API_v2.Repositorys.IRepositorys
{
    public interface IProjectRepository
    {
        Project? GetById(Guid id);
        Project Add(Project project);
        void Delete(Project project);
        List<Project> GetProjectsByUserId(Guid userId);
        List<ProjectMember> GetProjectMembers(Guid projectId);
        ProjectMember? GetMember(Guid projectId, Guid userId);
        void AddMember(ProjectMember member);
        void RemoveMember(ProjectMember member);
        void Save();
    }
}
